From: Sylvain <sylvain.becker@gmail.com>
Date: Sat, 19 Mar 2022 20:40:28 +0100
Subject: More integer overflow (see bug #187)

Make sure that 'width + alignment' doesn't overflow, otherwise
it could create a SDL_Surface of 'width' but with wrong 'pitch'

Bug: https://github.com/libsdl-org/SDL_ttf/issues/187
Bug-CVE: https://security-tracker.debian.org/tracker/CVE-2022-27470
Bug-Debian: https://bugs.debian.org/1010671
Origin: upstream, 2.19.0, commit:db1b41ab8bde6723c24b866e466cad78c2fa0448
---
 SDL_ttf.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/SDL_ttf.c b/SDL_ttf.c
index bf06b58..0f3c611 100644
--- a/SDL_ttf.c
+++ b/SDL_ttf.c
@@ -1248,7 +1248,7 @@ static SDL_Surface* Create_Surface_Solid(int width, int height, SDL_Color fg, Ui
      */
     void *pixels, *ptr;
     /* Worse case at the end of line pulling 'alignment' extra blank pixels */
-    Sint64 pitch = width + alignment;
+    Sint64 pitch = (Sint64)width + (Sint64)alignment;
     pitch += alignment;
     pitch &= ~alignment;
     size = height * pitch + sizeof (void *) + alignment;
@@ -1312,7 +1312,7 @@ static SDL_Surface* Create_Surface_Shaded(int width, int height, SDL_Color fg, S
      */
     void *pixels, *ptr;
     /* Worse case at the end of line pulling 'alignment' extra blank pixels */
-    Sint64 pitch = width + alignment;
+    Sint64 pitch = (Sint64)width + (Sint64)alignment;
     pitch += alignment;
     pitch &= ~alignment;
     size = height * pitch + sizeof (void *) + alignment;
@@ -1409,7 +1409,7 @@ static SDL_Surface *Create_Surface_Blended(int width, int height, SDL_Color fg,
         Sint64 size;
         void *pixels, *ptr;
         /* Worse case at the end of line pulling 'alignment' extra blank pixels */
-        Sint64 pitch = (width + alignment) * 4;
+        Sint64 pitch = ((Sint64)width + (Sint64)alignment) * 4;
         pitch += alignment;
         pitch &= ~alignment;
         size = height * pitch + sizeof (void *) + alignment;
